instrumentation/httpasyncclient/src/test/java/brave/httpasyncclient/ITTracingHttpAsyncClientBuilder.java

```diff
@@ -92,4 +99,52 @@ protected void post(CloseableHttpAsyncClient client, String pathIncludingQuery,
     
     takeSpan();
   }
+
+  @Test public void currentTraceContextIsVisibleInCallbackThread() throws Exception {
+    Tracer tracer = httpTracing.tracing().tracer();
+    AtomicBoolean callbackHasTraceCtx = new AtomicBoolean(false);
+    AtomicBoolean callbackCompleted = new AtomicBoolean(false);
+    server.enqueue(new MockResponse());
+    closeClient(client);
+    client = TracingHttpAsyncClientBuilder.create(httpTracing).build();
+    client.start();
+    ScopedSpan span = tracer.startScopedSpan("test");
+    TraceContext expectedTraceCtx = span.context();
+
+    try { 
```

Comment:
I think what's left here (besides nit on formatting) is to simplify the code and remove the wait loop. For example, if we use a countdownlatch instead of AtomicBoolean for callbackCompleted. You can search for references to CountdownLatch in brave for similar examples.

next, the assertion in 117 may actually occur off the main thread, which is why you are saving a boolean there. I think it is a bit cleaner to have an atomic reference of TraceContext, then you can do assertThat(callbackTraceContext.get()).isSameAs(expected) right after your countdown latch is done.